server/src/controllers/authentication.ts

```diff
@@ -0,0 +1,22 @@
+import { Request, Response } from 'express';
+import { compare } from 'bcrypt';
+import { User } from '../models';
+import { loginValidation } from '../validation';
+import { generateToken, serverErrs } from '../helpers';
+
+const login = async (req: Request, res: Response) => {
+  const { email, password } = req.body;
+  await loginValidation.validate({ email, password });
+
+  const user = await User.findOne({ where: { email } });
+  if (!user) throw serverErrs.BAD_REQUEST('Wrong Email Or Password');
+  const result = await compare(password, user.password);
+
+  if (!result) throw serverErrs.BAD_REQUEST('Wrong Email Or Password');
+
+  const { name, role, id } = user;
+  const token = await generateToken({ userID: id, role, name });
+  res.cookie('token', token);
+  return { status: 200, msg: 'logged in successfully' };
```

Comment:
I think we will need to return values needed to be set at `UserContext`, For now we will leave it and we can fix when we send request at login page